src/pages/workspace/FeatureEnabledAccessOrNotFoundWrapper.tsx

```diff
@@ -33,9 +35,25 @@ type FeatureEnabledAccessOrNotFoundComponentProps = FeatureEnabledAccessOrNotFou
 };
 
 function FeatureEnabledAccessOrNotFoundComponent(props: FeatureEnabledAccessOrNotFoundComponentProps) {
+    const pendingField = props.policy?.pendingFields?.[props.featureName];
     const isPolicyIDInRoute = !!props.policyID?.length;
+    const isFeatureEnabled = PolicyUtils.isPolicyFeatureEnabled(props.policy, props.featureName);
+
+    const [isPolicyFeatureEnabled, setIsPolicyFeatureEnabled] = useState(isFeatureEnabled);
+    const isFocused = useIsFocused();
+    const {isOffline} = useNetwork();
+
+    const shouldShowNotFoundPage = isEmptyObject(props.policy) || !props.policy?.id || !isPolicyFeatureEnabled;
     const shouldShowFullScreenLoadingIndicator = props.isLoadingReportData !== false && (!Object.entries(props.policy ?? {}).length || !props.policy?.id);
-    const shouldShowNotFoundPage = isEmptyObject(props.policy) || !props.policy?.id || !PolicyUtils.isPolicyFeatureEnabled(props.policy, props.featureName);
+
+    useEffect(() => {
+        if (!isFocused) {
+            return;
```

Comment:
@ZhenjaHorbach Can you elaborate about this one? 
This one will cause regression, because in some feature screens, we still have the modal screen, ie: CategoryDetail.
So if we opened CategoryDetail screen and disabled the category feature, we will only see the NotFoundScreen after we dismiss the modal.

Reply:
Evidence:

https://github.com/Expensify/App/assets/16502320/29326c89-dfae-4088-b55a-8fe6f533430b

Reply:
Still didn't work, please make sure you test it carefully.


https://github.com/Expensify/App/assets/16502320/db598b58-4bfa-4164-ba13-a885116282a3

Reply:
I mentioned few test cases already, please make sure it will test all these cases.
https://github.com/Expensify/App/issues/39443#issuecomment-2041401686

Reply:
Looks good!